engine/lite/runtime/src/test/scala/pl/touk/nussknacker/engine/lite/InterpreterTestRunnerTest.scala

```diff
@@ -102,4 +102,24 @@ class InterpreterTestRunnerTest extends AnyFunSuite with Matchers {
     results.externalInvocationResults("end") shouldBe List(ExternalInvocationResult("some-ctx-id", "end", 120))
   }
 
+  test("should handle fragment test parameters in test") {
+    val fragment = ScenarioBuilder
+      .fragment("fragment1", "in" -> classOf[String])
+      .filter("filter", "#in != 'stop'")
+      .fragmentOutput("fragmentEnd", "output", "out" -> "#in")
+
+    val parameterExpressions: Map[String, Expression] = Map(
+      "in" -> Expression("spel", "'some-text-id'")
+    )
+    val scenarioTestData = ScenarioTestData("fragment1", parameterExpressions)
+    val results = sample.test(fragment, scenarioTestData)
+
+    results.nodeResults("fragment1") shouldBe List(
+      NodeResult(ResultContext("fragment1", Map("in" -> "some-text-id")))
+    )
+    //TODO - why FragmentOutputDefinition is not handled well in tests? :(
```

Comment:
I suppose that it may happen that way because, `PartSubGraphCompiler` handles `FragmentOutputDefinition` as a `Sink` (see line 120 and 129), and `ProcessSplitter` split `Sink` as a separate part. I'm not sure but I remember that there were some tricks around that. Have you tried to follow the debugger in our `InterpreterInternal` and check why debugger doesn't enter `ProcessListener.nodeEntered`? AFAIK this method is responsible for collecting testing results.

Reply:
I was able to make it work, thanks for a hint - it helped :)